--- a/vcl/unx/gtk3/gdi/gtk3salnativewidgets-gtk.cxx-old	2012-04-22 16:06:51.611916299 +0000
+++ b/vcl/unx/gtk3/gdi/gtk3salnativewidgets-gtk.cxx	2012-04-22 16:07:57.005983725 +0000
@@ -622,7 +622,7 @@
     {
         if ( !pSpinVal )
         {
-            std::fprintf( stderr, "Tried to draw CTRL_SPINBUTTONS, but the SpinButtons data structure didn't exist!\n" );
+            fprintf( stderr, "Tried to draw CTRL_SPINBUTTONS, but the SpinButtons data structure didn't exist!\n" );
             return;
         }
         areaRect = pSpinVal->maUpperRect;
@@ -1310,10 +1310,10 @@
     }
     aInfo.m_eWidth = WIDTH_ULTRA_CONDENSED;
 
-    std::fprintf( stderr, "font name BEFORE system match: \"%s\"\n", aFamily.getStr() );
+    fprintf( stderr, "font name BEFORE system match: \"%s\"\n", aFamily.getStr() );
     // match font to e.g. resolve "Sans"
     psp::PrintFontManager::get().matchFont( aInfo, rSettings.GetUILocale() );
-    std::fprintf( stderr, "font match %s, name AFTER: \"%s\"\n",
+    fprintf( stderr, "font match %s, name AFTER: \"%s\"\n",
                   aInfo.m_nID != 0 ? "succeeded" : "failed",
                   rtl::OUStringToOString( aInfo.m_aStyleName, RTL_TEXTENCODING_ISO_8859_1 ).getStr() );
 
@@ -1422,7 +1422,7 @@
     gchar* pThemeName = NULL;
     g_object_get( pSettings, "gtk-theme-name", &pThemeName, (char *)NULL );
     #if OSL_DEBUG_LEVEL > 1
-    std::fprintf( stderr, "Theme name is \"%s\"\n", pThemeName );
+    fprintf( stderr, "Theme name is \"%s\"\n", pThemeName );
     #endif
 
 
