--- libreoffice-4.3.3.orig/sw/source/core/text/porfld.cxx
+++ libreoffice-4.3.3/sw/source/core/text/porfld.cxx
@@ -744,7 +744,7 @@
 SwGrfNumPortion::SwGrfNumPortion(
         SwFrm*,
         const OUString& rGraphicFollowedBy,
-        const SvxBrushItem* pGrfBrush,
+        const SvxBrushItem* pGrfBrush, OUString const & referer,
         const SwFmtVertOrient* pGrfOrient, const Size& rGrfSize,
         const bool bLft, const bool bCntr, const KSHORT nMinDst,
         const bool bLabelAlignmentPosAndSpaceModeActive ) :
@@ -758,7 +758,7 @@
     if( pGrfBrush )
     {
         *pBrush = *pGrfBrush;
-        const Graphic* pGraph = pGrfBrush->GetGraphic();
+        const Graphic* pGraph = pGrfBrush->GetGraphic(referer);
         if( pGraph )
             SetAnimated( pGraph->IsAnimated() );
         else
--- libreoffice-4.3.3.orig/sw/source/core/text/porfld.hxx
+++ libreoffice-4.3.3/sw/source/core/text/porfld.hxx
@@ -172,6 +172,7 @@
     SwGrfNumPortion( SwFrm *pFrm,
                      const OUString& rGraphicFollowedBy,
                      const SvxBrushItem* pGrfBrush,
+                     OUString const & referer,
                      const SwFmtVertOrient* pGrfOrient,
                      const Size& rGrfSize,
                      const bool bLeft,
--- libreoffice-4.3.3.orig/sw/source/core/text/txtfld.cxx
+++ libreoffice-4.3.3/sw/source/core/text/txtfld.cxx
@@ -51,6 +51,7 @@
 #include "reffld.hxx"
 #include "flddat.hxx"
 #include "fmtautofmt.hxx"
+#include <sfx2/docfile.hxx>
 
 static bool lcl_IsInBody( SwFrm *pFrm )
 {
@@ -459,9 +460,18 @@
 
         if( SVX_NUM_BITMAP == rNumFmt.GetNumberingType() )
         {
+            OUString referer;
+            if (auto const sh1 = rInf.GetVsh()) {
+                if (auto const doc = sh1->GetDoc()) {
+                    auto const sh2 = doc->GetPersist();
+                    if (sh2 != nullptr && sh2->HasName()) {
+                        referer = sh2->GetMedium()->GetName();
+                    }
+                }
+            }
             pRet = new SwGrfNumPortion( (SwFrm*)GetTxtFrm(),
                                         pTxtNd->GetLabelFollowedBy(),
-                                        rNumFmt.GetBrush(),
+                                        rNumFmt.GetBrush(), referer,
                                         rNumFmt.GetGraphicOrientation(),
                                         rNumFmt.GetGraphicSize(),
                                         bLeft, bCenter, nMinDist,
--- libreoffice-4.3.3.orig/sw/source/core/unocore/unosett.cxx
+++ libreoffice-4.3.3/sw/source/core/unocore/unosett.cxx
@@ -53,6 +53,7 @@
 #include <vcl/font.hxx>
 #include <editeng/flstitem.hxx>
 #include <vcl/metric.hxx>
+#include <sfx2/docfile.hxx>
 #include <svtools/ctrltool.hxx>
 #include <osl/mutex.hxx>
 #include <vcl/svapp.hxx>
@@ -1498,10 +1499,18 @@
             pData = new PropValData((void*)&aUString, UNO_NAME_GRAPHIC_URL, ::cppu::UnoType<OUString>::get());
             aPropertyValues.push_back(pData);
 
+            OUString referer;
+            if (pDoc != nullptr) {
+                auto const sh = pDoc->GetPersist();
+                if (sh != nullptr && sh->HasName()) {
+                    referer = sh->GetMedium()->GetName();
+                }
+            }
+
             //graphicbitmap
             const Graphic* pGraphic = 0;
             if(pBrush )
-                pGraphic = pBrush->GetGraphic();
+                pGraphic = pBrush->GetGraphic(referer);
             if(pGraphic)
             {
                 uno::Reference<awt::XBitmap> xBmp = VCLUnoHelper::CreateBitmap( pGraphic->GetBitmapEx() );
