diff --git a/configure.ac b/configure.ac
index 0241d92..7e521ee 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7463,7 +7463,7 @@ no|disable)
         # Unset variables set by the above AM_PATH_PYTHON so that
         # we actually do check anew.
         unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
-        AM_PATH_PYTHON([3.3],, [:])
+        AM_PATH_PYTHON([2.6],, [:])
         if test "$PYTHON" = :; then
             AC_MSG_RESULT([internal])
             enable_python=internal
@@ -7517,7 +7517,7 @@ if test $enable_python = system; then
         # we actually do check anew.
         unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
         # This causes an error if no python command is found
-        AM_PATH_PYTHON([3.3])
+        AM_PATH_PYTHON([2.6])
         python_include=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('INCLUDEPY'));"`
         python_version=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('VERSION'));"`
         python_libs=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('LIBS'));"`
diff --git a/unotest/source/python/org/libreoffice/unotest.py b/unotest/source/python/org/libreoffice/unotest.py
index 3ec016a..1df20f1 100644
--- a/unotest/source/python/org/libreoffice/unotest.py
+++ b/unotest/source/python/org/libreoffice/unotest.py
@@ -7,6 +7,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
 
+from __future__ import print_function
 import subprocess
 import time
 import uuid
