--- a/configure.ac-old	2012-03-02 21:48:47.000000000 +0000
+++ b/configure.ac	2012-03-02 21:51:29.000000000 +0000
@@ -9681,6 +9687,11 @@ AC_SUBST(GSTREAMER_0_10_CFLAGS)
 fi
 AC_SUBST(ENABLE_VLC)
 
+if test -x /usr/bin/dpkg-buildflags; then
+	HARDEN_CPPFLAGS=`dpkg-buildflags --get CPPFLAGS 2>/dev/null`
+fi
+AC_SUBST(HARDEN_CPPFLAGS)
+
 dnl ===================================================================
 dnl Check whether the OpenGL libraries are available
 dnl ===================================================================
--- a/config_host.mk.in-old	2012-04-06 14:17:55.164239390 +0200
+++ b/config_host.mk.in	2012-04-06 14:18:55.440239065 +0200
@@ -70,6 +70,7 @@
 export CPPUNIT_LIBS=$(gb_SPACE)@CPPUNIT_LIBS@
 export CPUNAME=@CPUNAME@
 export CDEFAULTOPT=@CDEFAULTOPT@
+export HARDEN_CPPFLAGS=@HARDEN_CPPFLAGS@
 export CROSS_COMPILING=@CROSS_COMPILING@
 export CURL=@CURL@
 export CURL_CFLAGS=$(gb_SPACE)@CURL_CFLAGS@
diff --git a/solenv/gbuild/gbuild.mk b/solenv/gbuild/gbuild.mk
index c71cb40..f826d65 100644
--- a/solenv/gbuild/gbuild.mk
+++ b/solenv/gbuild/gbuild.mk
@@ -255,6 +255,8 @@ gb_GLOBALDEFS += \
 		ENABLE_LTO \
 	)
 
+gb_GLOBALDEFS += $(HARDEN_CPPFLAGS)
+
 gb_GLOBALDEFS := $(sort $(gb_GLOBALDEFS))
 
 include $(GBUILDDIR)/Deliver.mk
