For whatever reason we get undefined referenced to g_module_*.

--- a/vcl/Library_vclplug_gtk3.mk-old	2012-04-01 23:57:18.435192769 +0200
+++ b/vcl/Library_vclplug_gtk3.mk	2012-04-02 00:07:03.610693627 +0200
@@ -55,6 +55,7 @@
 $(eval $(call gb_Library_add_libs,vclplug_gtk3,\
     $$(GTK3_LIBS) \
     $$(GTHREAD_LIBS) \
+    $$(GMODULE_LIBS) \
 ))
 
 $(eval $(call gb_Library_add_linked_libs,vclplug_gtk3,\
--- a/configure.in-old	2012-04-01 23:57:57.091193585 +0200
+++ a/configure.in	2012-04-01 23:58:51.926693332 +0200
@@ -6996,10 +6996,13 @@
     else
         AC_MSG_ERROR([gtk3 libraries of the correct versions, not found])
     fi
+    PKG_CHECK_MODULES(GMODULE, gmodule-2.0,,AC_MSG_ERROR([requirements to build the gtk3-plugin not met. Use --disable-gtk3 or install the missing packages]))
 fi
 AC_SUBST(GTK3_LIBS)
 AC_SUBST(GTK3_CFLAGS)
 AC_SUBST(ENABLE_GTK3)
+AC_SUBST(GMODULE_LIBS)
+AC_SUBST(GMODULE_CFLAGS)
 
 AC_MSG_CHECKING([which VCLplugs shall be built])
 ENABLE_GTK=""
--- a/set_soenv.in-old	2012-04-02 00:02:17.578693138 +0200
+++ b/set_soenv.in	2012-04-02 00:02:52.858693628 +0200
@@ -1486,6 +1486,8 @@
 ToFile( "ENABLE_GTK3",       "@ENABLE_GTK3@",      "e" );
 ToFile( "GTK3_CFLAGS",       "@GTK3_CFLAGS@",      "e" );
 ToFile( "GTK3_LIBS",         "@GTK3_LIBS@",        "e" );
+ToFile( "GMODULE_CFLAGS",    "@GMODULE_CFLAGS@",   "e" );
+ToFile( "GMODULE_LIBS",      "@GMODULE_LIBS@",     "e" );
 ToFile( "GTHREAD_CFLAGS",    "@GTHREAD_CFLAGS@",   "e" );
 ToFile( "GTHREAD_LIBS",      "@GTHREAD_LIBS@",     "e" );
 ToFile( "ENABLE_SYSTRAY_GTK", "@ENABLE_SYSTRAY_GTK@", "e" );
